driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -455,10 +463,10 @@ private void translateOptions(final Map<String, List<String>> optionsMap) {
                 connectTimeout = parseInteger(value, "connecttimeoutms");
             } else if (key.equals("sockettimeoutms")) {
                 socketTimeout = parseInteger(value, "sockettimeoutms");
-            } else if (key.equals("sslinvalidhostnameallowed") && parseBoolean(value, "sslinvalidhostnameallowed")) {
-                sslInvalidHostnameAllowed = true;
-            } else if (key.equals("ssl") && parseBoolean(value, "ssl")) {
-                sslEnabled = true;
+            } else if (key.equals("sslinvalidhostnameallowed")) {
+                sslInvalidHostnameAllowed = parseBoolean(value, "sslinvalidhostnameallowed");
```

Comment:
The change for the boolean params allow tests to assert that the values of the these options were set even when they are set to their default values.

Reply:
Ack